src/main/java/io/github/syst3ms/skriptparser/expressions/ExecExprListOperators.java

```diff
@@ -0,0 +1,212 @@
+				break;
+			case 3:
+				type = 3;
+				list = (Expression<Object>) expressions[0];
+				switch (parseContext.getParseMark()) {
+					case 0:
+						lower = (Expression<BigInteger>) expressions[1];
+						upper = (Expression<BigInteger>) expressions[2];
+						if (expressions.length == 4)
+							step = (Expression<BigInteger>) expressions[3];
+						break;
+					case 1:
+						lower = (Expression<BigInteger>) expressions[1];
+						if (expressions.length == 3)
+							step = (Expression<BigInteger>) expressions[2];
+						break;
+					case 2:
+						upper = (Expression<BigInteger>) expressions[1];
+						if (expressions.length == 3)
+							step = (Expression<BigInteger>) expressions[2];
+						break;
+					default:
+						throw new IllegalStateException();
+				}
+		}
+		var logger = parseContext.getLogger();
+		if (list.acceptsChange(ChangeMode.SET).isEmpty()) {
+			logger.error(
+					list.toString(null, logger.isDebug()) + " is static and cannot be changed",
+					ErrorType.SEMANTIC_ERROR
+			);
+			return false;
+		} else if (list.isSingle()) {
+			logger.error(
+					list.toString(null, logger.isDebug()) + " represents a single value",
+					ErrorType.SEMANTIC_ERROR,
+					"List operators only work on multiple objects at the same time"
+			);
+			return false;
+		}
+		return true;
+	}
+
+	@Override
+	public Object[] getValues(TriggerContext ctx) {
+		var values = list.getValues(ctx);
+		if (values.length == 0)
+			return new Object[0];
+
+		switch (type) {
+			case 0:
+				list.change(ctx, Arrays.copyOfRange(values, 0, values.length - 1), ChangeMode.SET);
+				return new Object[] {values[values.length - 1]};
+			case 1:
+				list.change(ctx, Arrays.copyOfRange(values, 1, values.length), ChangeMode.SET);
+				return new Object[] {values[0]};
+			case 2:
+				int ind = index.getSingle(ctx)
+						.filter(n -> n.signum() > 0 && n.compareTo(BigInteger.valueOf(values.length)) <= 0)
+						.map(n -> n.intValue() - 1)
+						.orElse(-1);
+				if (ind == -1) {
+					return new Object[0];
+				}
+				var skipped = new Object[values.length - 1];
+				for (int i = 0, k = 0; i < values.length; i++) {
+					if (i == ind) {
+						continue;
+					}
+					skipped[k++] = values[i];
+				}
+				list.change(ctx, skipped, ChangeMode.SET);
+				return new Object[] {values[ind]};
+			case 3:
+				var low = lower != null
+						? lower.getSingle(ctx).filter(n -> n.signum() > 0).map(n -> n.intValue() - 1).orElse(-1)
+						: 0;
+				var up = upper != null
+						? upper.getSingle(ctx).filter(n -> n.compareTo(BigInteger.valueOf(values.length)) <= 0).map(BigInteger::intValue).orElse(values.length)
+						: values.length;
+				var st = step != null
+						? step.getSingle(ctx).filter(n -> n.signum() != 0 && n.compareTo(BigInteger.valueOf(-values.length)) >= 0 && n.compareTo(BigInteger.valueOf(values.length)) <= 0).map(BigInteger::intValue).orElse(0)
+						: 1;
+
+				if (st < 0) {
+					CollectionUtils.reverseArray(values);
```

Comment:
Sorry, unfortunately that doesn't work. Take the list `a = [0,1,2,3,4,5]`.
Using python syntax for splicing, `a[4:1:-2] = [4,2]`. However, reversing the list gives `b = [5,4,3,2,1,0]`, and `b[0:5:2] = [5,3,1]`. I may be wrong about this due to indexing shenanigans, but please test it and make sure it works for larger samples either way.

This can probably be solved using some dirty modulo stuff, but honestly you'd be better off just trying to start at the given index, move by `st` every time, and check if you're still in the bounds at each step.

Reply:
In the lines under that statement I normally switched the lower and upper bound to provide the behavior you just specified. 
I'll test this again tomorrow but I'm fairly sure I did it right, because I tested this myself before pushing. But we'll see.

Reply:
Tested it, it works as intended. The only difference is that you'll need to use `splice {list::*} from 5 to 2 with step -2` to have the result in your example, because:
1) Indices in Skript start at 1.
2) The upper bound is inclusive.